src/components/modal/modal.tsx

```diff
@@ -71,22 +80,51 @@ export const Modal = (props: ModalProps) => {
   return ReactDOM.createPortal(<ModalContent {...props} />, document.getElementById('modal')!);
 };
 
-interface ModalHeaderProps {
-  children?: React.ReactNode;
-}
+type ModalHeaderProps =
+  | {
+      children: string;
+      tooltipText?: null;
+      size?: null;
+      noMargin?: boolean;
+    }
+  | {
+      children: string;
+      tooltipText: string;
+      size: 'large';
+      noMargin?: boolean;
```

Comment:
It doesn't seem we are using this variant of `ModalHeader` anywhere. Can't we just drop it?

Reply:
No, it's actually there in preparation for the upcoming `New Proposal` modal 😊

Reply:
Oh, right!